### PR TITLE
Fix the MKS_MINI_12864 Glitches on  FAST MCU and improve BootScreens Funtionality over LCP1768

### PR DESCRIPTION
**Pull request by DrDitto**
### Requirements

* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.

### Description
Fix all MKS_MINI_12864 glitches on FAST MCU. Also recover the BOOT_SCREEN and CUSTOM_BOOTSCREEN over SKR V1.1 and SKR V1.3. This Fix should fix other issues reported at LPC1768 and MKS_MINI_12864
My own test over SKR V1.1 and SKR V1.3 and MKS Sbase V1.3 was successfull and show no extrange screen errors or glitches .

### Benefits
The prosed changes, restore the broken  BOOT_SCREEN and CUSTOM_BOOTSCREEN over SKR V1.1 and SKR V1.3. Resolving other MKS_MINI_12864 glitches resported 

### Related Issues
None detected issues yet

### CODE REVIEW

**Comment by gloomyandy** [comment]
Your change to ultralcd_DOGM.h:
```C
  #if ENABLED(MKS_MINI_12864)                           // 8 stripes
    #define FORCE_SOFT_SPI
    #define U8G_CLASS U8GLIB_MINI12864_2X_HAL
  #else
    #define U8G_CLASS U8GLIB_MINI12864_2X
  #endif
  #define U8G_PARAM DOGLCD_CS, DOGLCD_A0                              // 8 stripes	  #define U8G_PARAM DOGLCD_CS, DOGLCD_A0    
```
will force this change on all users of the display not just those with the SKR boards. Unless you have tested this change on many configurations it is probably not a good idea to do this. At the very least you can move the setting of FORCE_SOFT_SPI into the pins file and perhaps trigger the other change you need based on the setting of that define (take a look at how the FYSETC_MINI_12864 handles a similar change).

File: Marlin/src/lcd/ultralcd.cpp [hunk 1]
```diff
@@ -229,7 +229,12 @@ millis_t next_button_update_ms;
 #endif // HAS_LCD_MENU
 
 void MarlinUI::init() {
-
+  //DrDitto
+  //BootScreen and Custom Boot screen moved
+  //Prior the LCD INIT
+  #if HAS_SPI_LCD && ENABLED(SHOW_BOOTSCREEN)
+    ui.show_bootscreen();
+  #endif
```
Comment (AnHardt) [inline Marlin/src/lcd/ultralcd.cpp:237]: How can you display the bootscreen before the display is initialized?

**Review by gloomyandy — COMMENTED** [review]
I'm pretty sure there are #defines for all of those magic hex numbers being used. But more importantly they seem to be setting similar settings to the one time initialisation code. Is it really needed to do all of this every time you update the display?

**Comment by DrDitto** [comment]
The Init Sequence in the HAL reset the screen so The first two images are
displayed outside of the U8g loop

**Comment by DrDitto** [comment]
New change insolate the change to LPC1768

**Comment by jmz52** [comment]
Global `#define FORCE_SOFT_SPI` is not a good choice.
STM32 systems uses shared hardware SPI to communicate with number of peripheral devices, not just MKS MINI 12864.
SD card, MAX6685 and touch screen controller will not appreciate GPIO mode on hardware SPI pins.

Also the LPC1768  is not the only platform affected by 'fact MCU' issue - https://github.com/MarlinFirmware/Marlin/issues/13550

**Comment by gloomyandy** [comment]
It seems there is some sort of underlying issue here causing the screen update problems. It would be good to get the the actual cause of the problem and fix that. I don't have this display so can't really test it at all @Bob-the-Kuhn do you have the hardware for this (I know you have done a lot of work on the Fysetc display, so probably understand the hardware/software well!)? 

@jmz52 on the STM32 did you need to change things to use the U8GLIB_MINI12864_2X_HAL routine? Would that change work on the STM32?

**Comment by DrDitto** [comment]
Ok, I make the changes in order to Insolate the change to LPC1768, this because I own SKR V1.1 SKR V1.3 and MKS SBASE V1.3 board to make the test procedures. And This changes works in all mentioned boards.

**Comment by DrDitto** [comment]
Is there any suggestion about this change, Tevo flash Using SKR v1.1, tevo tornado using SKR V1.3 and Tevo tarantula using MKS SBase V1.3 has more than 24 Hours using the change without any screen glitches

**Comment by thinkyhead** [comment]
Your repository has gone away. You'll need to close this and submit a new PR.

**Comment by DrDitto** [comment]
Ok

**Comment by DrDitto** [comment]
Fixing minor issues

**Pull request closed without merging**